server/test/unit/vitalsTest/checkButtonHeartbeatTest.js

```diff
@@ -260,4 +296,150 @@ describe('vitals.js unit tests: checkButtonHeartbeat', () => {
       expect(db.updateButtonsSentVitalsAlerts).to.not.be.called
     })
   })
+
+  describe("when client that is sending vitals's two buttons that are sending vitals last seen exceeds the threshold and no alerts have been sent", () => {
+    beforeEach(async () => {
+      // Set up scenario where one client has two buttons that have just disconnected
+      this.client = factories.clientFactory({ isSendingVitals: true })
+      this.buttonA = buttonFactory({ id: 'fakeIdA', displayName: 'Unit A', sentVitalsAlertAt: null, isSendingVitals: true, client: this.client })
+      this.buttonB = buttonFactory({
+        id: 'fakeIdB',
+        displayName: 'Unit B',
+        buttonSerialNumber: 'AB12-23456',
+        sentVitalsAlertAt: null,
+        isSendingVitals: true,
+        client: this.client,
+      })
+      this.buttonsVitalA = buttonsVitalFactory({
+        createdAt: exceededThresholdTimestamp,
+        button: this.buttonA,
+      })
+      this.buttonsVitalB = buttonsVitalFactory({
+        createdAt: exceededThresholdTimestamp,
+        button: this.buttonB,
+      })
+      sandbox.stub(db, 'getRecentButtonsVitals').returns([this.buttonsVitalB, this.buttonsVitalA])
+      sandbox.stub(db, 'getDisconnectedGatewaysWithClient').returns([])
+    })
+
+    it('should log three Sentry messages', async () => {
+      await vitals.checkButtonHeartbeat()
+      expect(helpers.logSentry).to.be.calledThrice
```

Comment:
Thanks for asking about this Micah! The reason I had this in there is because I want to ensure that there's exactly 3 messages being logged to Sentry and no more than that. So this test case would fail if a 4th Sentry message was logged. This is important to test because we only want to log one "client message" per client, even when a client has multiple buttons disconnected. 

If we took this test case out and my code wasn't performing as expected - it could potentially log two "client messages" since there are two buttons that have disconnected. My other test cases would still pass, but I wouldn't know that I was logging multiple "client messages" (when I was only expecting to log one).

Reply:
Ah, makes sense! Thanks for explaining!